mlflow/utils/autologging_utils.py

```diff
@@ -66,10 +74,33 @@ def log_fn_args_as_params(fn, args, kwargs, unlogged=[]):  # pylint: disable=W01
     try_mlflow_log(mlflow.log_params, params_to_log)
 
 
+def _update_wrapper_extended(wrapper, wrapped):
+    """
+    Update a `wrapper` function to look like the `wrapped` function. This is an extension of
+    `functools.update_wrapper` that applies the docstring *and* signature of `wrapped` to
+    `wrapper`, producing a new function.
+
+    :return: A new function with the same implementation as `wrapper` and the same docstring
+             & signature as `wrapped`.
+    """
+    updated_wrapper = functools.update_wrapper(wrapper, wrapped)
+    # Assign the signature of the `wrapped` function to the updated wrapper function.
+    # Certain frameworks may disallow signature inspection, causing `inspect.signature()` to throw.
+    # One such example is the `tensorflow.estimator.Estimator.export_savedmodel()` function
+    try:
+        updated_wrapper.__signature__ = inspect.signature(wrapped)
+    except Exception:  # pylint: disable=broad-except
+        _logger.debug("Failed to restore original signature for wrapper around %s", wrapped)
```

Comment:
```suggestion
        _logger.warning("Failed to restore original signature for wrapper around %s", wrapped)
```

Should we use `warning` here?

Reply:
Because we expect this to happen in some cases (e.g., when patching `tf.estimator.Estimator.export_savedmodel()`), I think it's best not to emit a warning to users here. The debug log should help us for debugging purposes later on.